src/stream/ws/ws.c

```diff
@@ -97,9 +99,12 @@ static ssize_t stream_ws_process_ping(struct stream_ws* ws, size_t offset)
 
 	int payload_len = MIN(ws->read_index, ws->header.payload_length);
 
-	// Feed back the payload:
-	stream_tcp_send(&ws->base, rcbuf_from_mem(ws->read_buffer + offset,
-			payload_len), NULL, NULL);
+	// Feed back the (unmasked) payload:
+	rcbuf = rcbuf_new(malloc(payload_len), payload_len);
+	assert(rcbuf->payload);
```

Comment:
This should be `assert(rcbuf && rcbuf->payload)`.

Reply:
adjusted